core/unit_test/TestRealloc.hpp

```diff
@@ -51,6 +54,14 @@ void impl_testRealloc() {
     const int* newPointer = view_1d.data();
     EXPECT_EQ(oldPointer, newPointer);
   }
+  // Check #6992 fix (no default initialization in realloc without initializing)
```

Comment:
It's fine but noting that this test does not need to be instantiated multiple times like all others do.
It wouldn't hurt to add a comment observing how it contrasts with the other blocks.